<?xml version="1.0"?>

<gazebo:world 
  xmlns:xi="http://www.w3.org/2001/XInclude"
  xmlns:gazebo="http://playerstage.sourceforge.net/gazebo/xmlschema/#gz" 
  xmlns:model="http://playerstage.sourceforge.net/gazebo/xmlschema/#model" 
  xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor" 
  xmlns:window="http://playerstage.sourceforge.net/gazebo/xmlschema/#window" 
  xmlns:param="http://playerstage.sourceforge.net/gazebo/xmlschema/#param" 
  xmlns:body="http://playerstage.sourceforge.net/gazebo/xmlschema/#body" 
  xmlns:geom="http://playerstage.sourceforge.net/gazebo/xmlschema/#geom" 
  xmlns:joint="http://playerstage.sourceforge.net/gazebo/xmlschema/#joint" 
  xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface" 
  xmlns:ui="http://playerstage.sourceforge.net/gazebo/xmlschema/#ui"
  xmlns:rendering="http://playerstage.sourceforge.net/gazebo/xmlschema/#rendering" 
  xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller"
  xmlns:physics="http://playerstage.sourceforge.net/gazebo/xmlschema/#physics" >

  <verbosity>5</verbosity>

  <physics:ode>
    <stepTime>0.03</stepTime>
    <gravity>0 0 -9.8</gravity>
    <cfm>10e-5</cfm>
    <erp>0.3</erp>
  </physics:ode>

  <rendering:gui>
    <type>fltk</type>
    <size>640 480</size>
    <pos>0 0</pos>
  <frames>
      <row height="100%">
        <camera width="100%">
          <xyz>-0.5  1.5  0.5</xyz>
          <rpy> 0.0  0.0 -90</rpy>
        </camera>
      </row>
    </frames>

  </rendering:gui>

  <rendering:ogre>
    <ambient>0.4 0.4 0.4 1.0</ambient>
    <sky>
      <material>Gazebo/CloudySky</material>
    </sky>

    <!--<fog>
      <color>0.9 0.9 0.9</color>
      <linearStart>50</linearStart>
      <linearEnd>100</linearEnd>
    </fog>
    -->
  </rendering:ogre>

   <!-- Ground Plane -->
  <model:physical name="plane1_model">
    <xyz>0 0 0</xyz>
    <rpy>0 0 0</rpy>
    <static>true</static>

    <body:plane name="plane1_body">
      <geom:plane name="plane1_geom">
        <normal>0 0 1</normal>
        <size>2000 2000</size>
        <segments>10 10</segments>
        <uvTile>100 100</uvTile>
        <material>Gazebo/GrassFloor</material>
      </geom:plane>
    </body:plane>
  </model:physical>

  <!-- White Directional light -->
  <model:renderable name="directional_white">
    <light>
      <type>directional</type>
      <direction>0 -0.8 -0.3</direction>
      <diffuseColor>0.9 0.9 0.9</diffuseColor>
      <specularColor>0.0 0.0 0.0</specularColor>
      <attenuation>100 0.0 1.0 0.4</attenuation>
    </light>
  </model:renderable>

  <!-- Small barrel for manipulation testing -->
  <model:physical name="cylinder1_model">
    <xyz>-0.6 0 0.040</xyz>
    <rpy>0 0 0</rpy>
    <body:cylinder name="cylinder1_body">
      <geom:cylinder name="cylinder1_geom">
	      <size>0.015 0.080</size>
	      <mass>0.01</mass>
	      <visual>
	        <mesh>unit_cylinder</mesh>
	        <scale>0.030 0.030 0.080</scale>
	        <material>Gazebo/RustyBarrel</material>
	      </visual>
      </geom:cylinder>
    </body:cylinder>
  </model:physical>

  <!-- Pioneer with attached arm -->
  <model:physical name="pioneer2at_model1">
    <xyz>0 0 0.215</xyz>
    <rpy>0.0 0.0 0.0</rpy>

    <controller:differential_position2d name="controller1">
      <leftJoint>left_front_wheel_hinge</leftJoint>
      <rightJoint>right_front_wheel_hinge</rightJoint>
      <wheelSeparation>0.34</wheelSeparation>
      <wheelDiameter>0.15</wheelDiameter>
      <torque>10.0</torque>
      <interface:position name="position_iface_0"/>
    </controller:differential_position2d>

    <!-- Pioneer 5-dof Arm -->
    <model:physical name="pioneerarm">
      <xyz> -0.150  0.000  0.060</xyz>
      <rpy>    0.0    0.0  180.0</rpy>

      <attach>
        <parentBody>chassis_body</parentBody>
        <myBody>base_body</myBody>
      </attach>
<!--
       <model:physical name="cam1_model">
         <xyz>0 0 0</xyz>
         <rpy>0.0 0.0 90.0</rpy>
         <body:empty name="cam1_body">
           <sensor:camera name="cam1_sensor">
              <imageSize>640 480</imageSize>
              <controller:generic_camera name="cam1_controller">
                <interface:camera name="camera_iface_0"/>
              </controller:generic_camera>
            </sensor:camera>
          </body:empty>
        </model:physical>
-->

      <include embedded="true">
        <xi:include href="models/pioneerarm.model" />
      </include>
    </model:physical>


    <include embedded="true">
      <xi:include href="models/pioneer2at.model" />
    </include>
  </model:physical>


</gazebo:world>
